system-tests/tests/smoke/cre/helpers_test.go

```diff
@@ -267,20 +266,52 @@ func createAndFundAddresses(t *testing.T, testLogger zerolog.Logger, numberOfAdd
 		testLogger.Info().Msgf("Generated address #%d: %s", orderNum, addressToRead.Hex())
 
 		testLogger.Info().Msgf("Funding address '%s' with amount of '%s' wei", addressToRead.Hex(), amountToFund.String())
-		receipt, funErr := crefunding.SendFunds(t.Context(), testLogger, sethClient, crefunding.FundsToSend{
-			ToAddress:  addressToRead,
-			Amount:     amountToFund,
-			PrivateKey: sethClient.MustGetRootPrivateKey(),
-		})
-		require.NoError(t, funErr, "failed to send funds")
-		testLogger.Info().Msgf("Funds sent successfully to address '%s': txHash='%s'", addressToRead.Hex(), receipt.TxHash)
+
+		if bcOutput.BlockchainOutput.Family == "tron" {
+			if err := fundTronAddress(t, testLogger, addressToRead, amountToFund, bcOutput, fullCldEnvOutput); err != nil {
+				return nil, err
+			}
+		} else {
+			receipt, funErr := crefunding.SendFunds(t.Context(), testLogger, sethClient, crefunding.FundsToSend{
+				ToAddress:  addressToRead,
+				Amount:     amountToFund,
+				PrivateKey: sethClient.MustGetRootPrivateKey(),
+			})
+			require.NoError(t, funErr, "failed to send funds")
+			testLogger.Info().Msgf("Funds sent successfully to address '%s': txHash='%s'", addressToRead.Hex(), receipt.TxHash)
+		}
 
 		addressesToRead = append(addressesToRead, addressToRead)
 	}
 
 	return addressesToRead, nil
 }
 
+func fundTronAddress(t *testing.T, testLogger zerolog.Logger, addressToRead common.Address, amountToFund *big.Int, bcOutput *cre.WrappedBlockchainOutput, fullCldEnvOutput *cre.FullCLDEnvironmentOutput) error {
+	tronChains := fullCldEnvOutput.Environment.BlockChains.TronChains()
+	tronChain, exists := tronChains[bcOutput.ChainSelector]
+	if !exists {
+		return fmt.Errorf("TRON chain not found for selector %d", bcOutput.ChainSelector)
+	}
+
+	// Convert EVM address to Tron address format
+	receiverAddress := address.EVMAddressToAddress(addressToRead)
+	testLogger.Info().Msgf("TRON chain %d: Address conversion - EVM: %s -> Tron: %s", bcOutput.ChainSelector, addressToRead.Hex(), receiverAddress.String())
+
+	tx, err := tronChain.Client.Transfer(tronChain.Address, receiverAddress, amountToFund.Int64())
+	if err != nil {
+		return fmt.Errorf("failed to create transfer transaction for TRON address %s: %w", addressToRead.Hex(), err)
+	}
+
+	txInfo, err := tronChain.SendAndConfirm(t.Context(), tx, nil)
+	if err != nil {
+		return fmt.Errorf("failed to send and confirm transfer to TRON address %s: %w", addressToRead.Hex(), err)
+	}
+
+	testLogger.Info().Msgf("Successfully funded TRON address %s with %s SUN, txHash: %s", receiverAddress.String(), amountToFund.String(), txInfo.ID)
+	return nil
+}
```

Comment:
very similar to the function from `system-tests/lib/cre/environment/fund_cl_nodes_op.go`, maybe we could make that one more universal and just use it in the test?

Reply:
That feels out of scope for this PR, since I was just expanding on the existing functionality.

Reply:
how come? you added both functions in this PR

Reply:
The test was already using `crefunding.SendFunds` I just extracted it into a separate function. cl nodes op is specific to nodes in the don topology whereas this function is simply funding random addresses to be read by the workflow

Reply:
wait, you have added both `fundTronAddress` here and in `system-tests/lib/cre/environment/fund_cl_nodes_op.go`. Didn't you? I think you could use the function from `system-tests/tests/smoke/cre/helpers_test.go` to also fund nodes instead of doing a copy & pasta. All you'd need to do is pass node address to it.